### DISCUSSION TITLE
how to rectangle paste

### DISCUSSION THREAD

**Discussion by vincentaxhe** [helix-editor/helix#11945] [Q&A] [answered]
cannot use multi cursors, then paste, it will produce multi copies, or 'R' an rectangle selection, what's the right way to do it?

**Comment by TornaxO7** [comment]
Hi! Could you please give a little example of what you mean?

**Reply by vincentaxhe** [reply]
```
a
bc
def
g
hi
jkl
```
to
```
a  g
bc hi
defjkl
```
vertial or rectangle paste

**Reply by TornaxO7** [reply]
Ok, if I'm understanding you correctly, you'd like to move the three last lines:
```
g
hi
jkl
```
to the top three lines, but to the end of each line.

Well at least for me, I don't know if that's possible with multi-cursor, but I don't think it's possible. However, I'd probably do that with macros and formatting.

**Reply by vincentaxhe** [reply]
multi cursor or selection edit are not helpful, macros can do it which need more effort.
neovim can do it, it's a shame if helix really can not do any vertical paste. It should not be a obstacle.

**Reply by pierrelgol** [reply, marked as answer]
I can do it easily in helix.

1. put cursor at g
2. Shift-C 3x
3. vgl (go line end)
5. d
6. 3k
7. gl
8. p
9. i
10. Esc
11. Shift &

Edit I'm not sure this would work with the default config tho. But if you need my config I have it on my github, the only thing I think is the Esc for the collapse selection thing.

**Reply by vincentaxhe** [reply]
yeah, I notice vgl select string not include "\n", before I use 'x' select lines include "\n", when paste lines, It will produce multi copies, But if only strings helix know should append them.
It's a little subtle.
